### ISSUE TITLE
Reverse Proxy Authentication Clone Password

### ISSUE THREAD

**Issue by joe-french** [gogits/gogs#2889]
- Gogs version (or commit ref): 0.9.0.0306
- Git version: 1.8.3.1
- Operating system: CentOS 7
- Database:
  - [ ] PostgreSQL
  - [ ] MySQL
  - SQLite
- Can you reproduce the bug at http://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  -  Not relevant
- Log gist: 
```
2016/03/27 00:45:47 [D] Session ID: <Session ID Redacted>
2016/03/27 00:45:47 [D] CSRF Token: <CSRF Token Redacted>
2016/03/27 00:45:48 [...routers/repo/http.go:122 HTTP()] [E] invalid token
```

## Description
I believe there's an issue with authentication when doing a git clone / git pull / etc. on a private repository using HTTP, and a reverse proxy. 

Using apache basic auth, and reverse proxy I can login and browse gogs, but I can't clone a repository. Unless I set my gogs password to match the password in the htaccess file. 

From looking at the code in routers/repo/http.go there's a comment that looks awfully suspect around line 97 https://github.com/gogits/gogs/blob/master/routers/repo/http.go#L97-L100 .

Basically I believe git clones / pulls aren't honoring the `REVERSE_PROXY_AUTHENTICATION` parameters.

**Comment by BlackVoid** [comment]
You can solve it by setting an environment variable in the site config `SetEnv proxy-chain-auth On` which passes the information to Gogs.
